internal/controller/postgrescluster/controller.go

```diff
@@ -213,35 +210,11 @@ func (r *Reconciler) Reconcile(
 		return result, err
 	}
 
-	if config.RegistrationRequired() && !r.registrationValid() {
-		if !registrationRequiredStatusFound(cluster) {
-			addRegistrationRequiredStatus(cluster, r.PGOVersion)
-			return patchClusterStatus()
-		}
-
-		if r.tokenAuthenticationFailed() {
-			r.Recorder.Event(cluster, corev1.EventTypeWarning, "Token Authentication Failed", "See "+r.RegistrationURL+" for details.")
-		}
-
-		if shouldEncumberReconciliation(r.Registration.Authenticated, cluster, r.PGOVersion) {
-			emitEncumbranceWarning(cluster, r)
-			// Encumbrance is just an early return from the reconciliation loop.
-			return patchClusterStatus()
-		} else {
-			emitAdvanceWarning(cluster, r)
-		}
-	}
-
-	if config.RegistrationRequired() && r.registrationValid() {
-		if tokenRequiredConditionFound(cluster) {
-			meta.RemoveStatusCondition(&cluster.Status.Conditions, v1beta1.TokenRequired)
-		}
-
-		if registrationRequiredStatusFound(cluster) {
-			cluster.Status.RegistrationRequired = nil
-			r.Recorder.Event(cluster, corev1.EventTypeNormal, "Token Verified", "Thank you for registering your installation of Crunchy Postgres for Kubernetes.")
-		}
+	if r.Registration != nil && r.Registration.Required(r.Recorder, cluster, &cluster.Status.Conditions) {
+		registration.SetAdvanceWarning(r.Recorder, cluster, &cluster.Status.Conditions)
 	}
+	cluster.Status.RegistrationRequired = nil
+	cluster.Status.TokenRequired = ""
```

Comment:
Will these lines eventually go away?

Reply:
Eventually. I noticed some odd behavior with `managedFields` when removing these from the CRD. We should zero them out (while in the CRD) before removing them entirely.